### ISSUE TITLE
Newly created Content Types do not appear in the Content Types list

### ISSUE THREAD

**Issue by cesarvarela** [strapi/strapi#2213]
- 10.12.0
- 6.4.1
- 3.0.0-alpha.14.4.0
- 5.7.23
- Mojave

**What is the current behavior?**
Newly created types do not appear in the content types list.

**Steps to reproduce the problem**
Just follow the Quick Start guide

**What is the expected behavior?**
They should appear. I see them in the database and in the Content Manager.

**Suggested solutions**
🤷‍♂️

**Comment by abdonrd** [comment]
Same issue here! :)

**Comment by pbov** [comment]
For for debug reasons: Works for me on High Sierra with a fresh install (with npm)

- npm: 6.4.1
- node: v9.11.1
- strapi: 3.0.0-alpha.14.4.0
- mongodb 4.0.3
- macOS 10.13.6

**Comment by masiya255** [comment]
Does not work when upgrading from version 14.3 (

**Comment by jacobgelman** [comment]
Same issue. I'm running the same version of everything as cesarvarela.

**Comment by cesarvarela** [comment]
I've tried both mongo and mysql and got the same result. 

once thing I've noticed is that If I manually delete my user from the db and then signup again, I do see the new Types, but if I refresh the page the items disappear again...

**Comment by joakgord** [comment]
I have got the same issue as @cesarvarela

**Comment by stiko** [comment]
Same issue here. I had to roll back to 14.3

**Comment by cesarvarela** [comment]
I've found more stuff, maybe this can help you guys::

I can access the new types If I do what I mention above, but when I add new fields to this new types, they don't work immediately, the `Type.settings.json` gets modified with the new field definition, but only gets visible in the UI if I restart strapi. 

Is this a caching issue maybe?

**Comment by stiko** [comment]
All my content types disappear if I modify a content type from content type builder. I only get them visible if I clear cache from browser settings.

**Comment by orangesmartie92** [comment]
@stiko same here.

**Comment by mnlbox** [comment]
I have same issue. After upgrade to 14.4.0
My other issue (#2114) fixed but now I can't see my new content types in side bar.

**Comment by abdonrd** [comment]
> Hello @cesarvarela and @abdonrd.
> Did you install strapi with yarn ? or with npm ?
> 
> It work on my side, new fresh app with new fresh database.

With npm!

**Comment by mnlbox** [comment]
I also installed from npm based on this migration guide: https://github.com/strapi/strapi/issues/2213

**Comment by lauriejim** [comment]
Was it a app migration or new app ?

**Comment by mnlbox** [comment]
@lauriejim I test new app and also migrate an old app. Both have same issue.

**Comment by mnlbox** [comment]
@lauriejim I think it's based on browser. In chromium anything is OK but I have issue in Firefox. Is it a cache problem?

**Comment by orangesmartie92** [comment]
@mnlbox I also tried migrating to 14.3 but same issues too

**Comment by orangesmartie92** [comment]
> For for debug reasons: Works for me on High Sierra with a fresh install (with npm)
> 
> * npm: 6.4.1
> * node: v9.11.1
> * strapi: 3.0.0-alpha.14.4.0
> * mongodb 4.0.3
> * macOS 10.13.6

I have all the same versions for all of the above but my node version is v10.10.0.

It works on my mac, but when I try on a remote ubuntu 16.04, it doesn't work

**Comment by abdonrd** [comment]
> Was it a app migration or new app ?

I have tried both!

**Comment by davidenq** [comment]
Hi,
I've same issue, but after some steps I've achieved that the table for a new content type appears. 

How to reproduce it.

- First, I've run the application.
- Next, I've tried to create a new content type, then that issue appears (the table doesn't appear in database but all files inside api folder have been created). 
- After that, I've stopped the application and I've run again it. 
- Finally, now the table will appear for new content type (obviously, if you previously have tried to created that content type).

Also I can't see new content types in side bar. (It appears when I clear browser cache)

**Comment by soupette** [comment]
Well the problem seems to be with Firefox when running the build. The error doesn't appear in dev mode (on `localhost:4000`).

**Comment by abdonrd** [comment]
I always use Chrome. 😕

**Comment by davidenq** [comment]
I've been working with Chrome but, I have same issue.
Google Chrome Version 69.0.3497.100 (Official Build) (64-bit) on MacOS X El Capitan Version 10.11.6 (15G31)

**Comment by abdonrd** [comment]
I can't see the `/content-manager/models` request in the Network tab when the content doesn't appear.

**Comment by mnlbox** [comment]
@soupette Is it related to #1271 ?? :thinking: 
If it is I think it's better to fix both of them in #2229 PR. :wink:

**Comment by Dogtiti** [comment]
I have the same issue.
and I have tried many times to reinstall it with Quick Start guide，but it still doesn't appear in the content types list.

**Comment by soupette** [comment]
@mnlbox the fix I made should also fix #1271.
@Dogtiti there's a PR open #2229 until this is released you can just take the code and run the following command at the root of your project `npm run setup` your app should work properly after that.

**Comment by Dogtiti** [comment]
@soupette  thx.
however,I need it Immediately，I have took your code but I can't build them, so how can I do?
maybe I have built them ,but it didn't change any more, so what should be the right steps?###

**Comment by roghernandez** [comment]
> @mnlbox the fix I made should also fix #1271.
> @Dogtiti there's a PR open #2229 until this is released you can just take the code and run the following command at the root of your project `npm run setup` your app should work properly after that.

Hey @soupette :) 
I'm running 
node `v11.1.0`
strapi `3.0.0-alpha.14.4.0` 
Mac OSX 10.14.1 here,

When I run `npm run setup` and start the server again, I'm still not seeing that newly created content type.

**Comment by Globerada** [comment]
Same issue here. Itis strange, because my app deployed at Heroku shows the content types :S

BTW, at localhost, only users type is show.

**Comment by Gillinghammer** [comment]
Same issue, on Chrome.
Got it working though by pushing my changes to my production server - then added content to the content type and pulled my changes locally... now I have it in the side bar.

**Comment by merijnvanessen** [comment]
Hi,
![schermafbeelding 2018-11-14 om 11 32 51](https://user-images.githubusercontent.com/23075300/48477247-67453400-e801-11e8-81f7-0df4de104a19.png)
It's still there.

They appear when I clear my cookies / local storage. After the first login.
They disappear when I refresh

No errors in my console

**Comment by derrickmehaffy** [comment]
Related: https://github.com/strapi/strapi/issues/2306

@merijnvanessen can you post your comment on the new issue. Also can you please include a screenshot of the error in your browser console.

**Comment by soupette** [comment]
It's weird is that's it's working on my side in both Firefox and Chrome.

**Comment by derrickmehaffy** [comment]
@soupette like I said in 2306 I couldn't duplicate it either on Linux Mint in both Chrome Stable/Beta & Firefox stable

**Comment by kalms** [comment]
I'm also experiencing this. Version: 3.0.0-alpha.14.4.0

Oddly enough, I've seen it work and then disappear again upon login. I login and I can view the content types. Upon refresh they disappear.

<img width="1449" alt="screen shot 2018-11-19 at 11 59 02" src="https://user-images.githubusercontent.com/867757/48722746-c5ec2280-ebf2-11e8-83c5-e6d2205dd703.png">

**Comment by derrickmehaffy** [comment]
@kalms : https://github.com/strapi/strapi/releases/tag/v3.0.0-alpha.14.5

**Comment by kalms** [comment]
@derrickmehaffy Ah sorry. Carry on! 👍

**Comment by Dogtiti** [comment]
So，Has it been fixed already in the latest version?

**Comment by derrickmehaffy** [comment]
@Dogtiti the latest is v16, please test on that. I believe this issue is random and seem to be user environment based. Many of the loading functions of the adminUI have been modified to wait on the browser and I believe this issue was "indirectly" fixed due to that.

**Comment by coyr** [comment]
I have a similar issue. 3.0.0-alpha.21 
![image](https://user-images.githubusercontent.com/31460/51772225-7e35bb80-20b9-11e9-80cd-402369199640.png)
Is there some solution?

**Comment by lauriejim** [comment]
@coyr How did you start your Strapi application ? `npm start` or `strapi start` or using a module like `pm2` ?

**Comment by coyr** [comment]
@lauriejim When I use the development environment `strapi start` works fine. When I use pm2 `NODE_ENV=production pm2 start server.js --name api` now it is not listing the content types in content types page. Also when I click a content type in the sidebar and I try to add new content, it keeps loading
![image](https://user-images.githubusercontent.com/31460/52020256-a6a92580-24be-11e9-964f-129d38157cd6.png)

I tried to stop and delete the process, restart nginx and restart with `NODE_ENV=production pm2 start server.js --name api` but it doesn't work. What can I try?

**Comment by derrickmehaffy** [comment]
@coyr try deleting your `plugin_content-manager_schema` from `core_store` in your database then restart your Strapi server.

**Comment by coyr** [comment]
@derrickmehaffy it works! thanks

**Comment by nikosolihin** [comment]
@derrickmehaffy That works perfectly!

**Comment by coder477** [comment]
I have the same issue while creating and deleting content-types. tried both with mongo and mysql. But the issue is only when I am trying to run the app through pm2. The normal "strapi start" has no issues.

**Comment by coder477** [comment]
@derrickmehaffy  Yes I too have no issues with `strapi start `. Actually, the app would not be directly used by me, once I have launched it on a server, people will start using it and create content-types. I am using `NODE_ENV=development` with pm2, as I don't want this to be terminated in between. Any alternate If you could suggest.

**Comment by derrickmehaffy** [comment]
@coder477 Nope, strapi is designed so that it should not auto restart when deployed to a server. Hence the reason for the different environments. Only the `strapi start` command allows for it to auto restart IE when new content types are created. I would suggest to other members on your team to clone it locally for development and follow normal git commit and proper testing before deployment.

**Comment by derrickmehaffy** [comment]
Unlike other CMSes, Strapi is effective an API framework aka a "Headless" CMS. 😉

**Comment by markjlaney** [comment]
@soupette every time I delete this record - plugin_content-manager_schema - I have to upload all of my images again. Do you have a better workaround for this issue? The more content I build into my CMS, the more impossible this solution will become.

**Comment by derrickmehaffy** [comment]
@soupette @alexandrebodin to clarify I don't think the Upload multi-file should be stored in that content-manager schema entry as it is not just a UI thing but a backend one as well, I think that entry should be stored in the model settings JSON file.

**Comment by Fdom92** [comment]
Just wanted to post my issue in case some one have the same. I work with Strapi and Heroku i got same problems, on staging environment all was working good but on production get the loading spinner and on the content manager i didn't see the new content type.

I checked on mLab the `plugin_content-manager_schema` from `core_store` and i discover that i had two, i just removed both, redeploy and all is working good, so thanks for the fix @derrickmehaffy

**Comment by j27gj** [comment]
I delete data.db file inside .tmp folder and the issue was resolved.

**Comment by Nauman440** [comment]
I created a project with beta version and everything is working fine you can follow this article to create content types in your project.

https://www.mstweaks.com/how-to-create-content-types-and-fetch-data-through-api-in-strapi/

**Comment by DanielWahl** [comment]
I have the same issue after a new install. Adding ContentTypes is working, but they are not shown in the list. (Development Mode)
Restarting the Node-Server every time you add a ContentType will "fix" it. After the restart they are shown.
But this can't be a proper solution.

Does there exist a bugfix at the current date?

**Comment by alexandrebodin** [comment]
@DanielWahl This issue is really old and outdated. Can you please open a new one so we know your setup / versions etc. Thanks !

**Comment by Aryan-ki-codepanti** [comment]
> 

I am facing same issue . Can you give probable fixes?

**Comment by lack-of-gravitas** [comment]
> > 
> 
> I am facing same issue . Can you give probable fixes?

LOL good luck they'll probably tell you to upgrade your version -- strapi is quickly heading downhill imo

**Comment by sayoojt** [comment]
> > 
> 
> 
> 
> I am facing same issue . Can you give probable fixes?

Did you get any solution ?

Am using strapi 3.6.8, getting the similar issue, is there any row in core_store which I can check for the list of tables or any other cache in database ?

**Comment by markcmlee** [comment]
Same issues here, v3.6.8.
Happened suddenly with no prior occurences after adding a new collection.
Everything works locally and even in the first minutes of the deployed app, but the new connections and endpoints disappear

**Comment by derrickmehaffy** [comment]
Hello,

Thank you for reporting this potential bug report, please keep in mind that we are no longer accepting bug reports for v3 unless they could be deemed high or critical severity or related to security in nature.

The recommended action we suggest for v3 users is to utilize the various migration resources to upgrade your package from Strapi v3 to Strapi v4:

- [Code Migration Guide](https://docs.strapi.io/developer-docs/latest/update-migration-guides/migration-guides/v4/code-migration.html) and/or [Codemods project](https://github.com/strapi/codemods)
- [Data Migration Guide](https://docs.strapi.io/developer-docs/latest/update-migration-guides/migration-guides/v4/data-migration.html) and/or [Data Migration Scripts](https://github.com/strapi/migration-scripts)
- [Plugin Migration Guide](https://docs.strapi.io/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin-migration.html)

Please see our [Security file](https://github.com/strapi/strapi/security) for more information about supported versions.
For now I will mark this issue as closed as we do not believe this issue is Critical/High Severity nor related to security.

Thank You
